File: gateway-boot/src/main/java/com/codingas/gateway/domain/iam/entity/UserApiKey.java
package com.codingas.gateway.domain.iam.entity;

import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

/**
 * 用户 API Key 领域实体
 * <p>
 * 一个 Key 归属一个用户，通过用户所属团队继承渠道访问权限。
 * keyHash 用于认证验证，keyPlain 用于创建时传入和详情展示（由基础设施层加解密）。
 * </p>
 */
@Setter
@Getter
public class UserApiKey {

    private Long id;
    private Long userId;
    /**
     * -- GETTER --
     * Key 哈希（SHA-256），用于认证验证，由基础设施层在 save 时计算
     */
    private String keyHash;
    private String keyPrefix;
    /**
     * -- GETTER --
     * Key 明文，创建时设置，查询时由基础设施层解密填充
     */
    private String keyPlain;
    private String name;
    /** 是否已删除（逻辑删除） */
    private boolean deleted;
    private Instant createdAt;
    private Instant updatedAt;

    public UserApiKey() {
    }

    /** 检查密钥是否可用 */
    public boolean isAvailable() {
        return !deleted;
    }

    /**
     * 重写 toString 排除敏感字段
     */
    @Override
    public String toString() {
        return "UserApiKey{" +
                "id=" + id +
                ", userId=" + userId +
                ", keyPrefix='" + keyPrefix + '\'' +
                ", name='" + name + '\'' +
                ", deleted=" + deleted +
                '}';
    }
}

File: gateway-console/src/types/userApiKey.ts
/*
 * Copyright © 2025-2026 codingas.com
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * 用户 API Key 相关类型（与后端 UserApiKeyResponse 一致）
 *
 * 一个 Key 归属一个用户，并挂载到具体应用（applicationId）作为权限锚点——
 * 通过应用-渠道授权关系（ApplicationChannel）继承渠道访问权限。
 * applicationId 为 null 时权限路由返回空集，Key 不可用。
 */

/** 用户 API Key */
export interface UserApiKey {
  id: number;
  userId: number;
  applicationId: number | null;
  keyPrefix: string;
  keyPlain: string;
  name: string;
  deleted: boolean;
  createdAt: string;
  updatedAt: string;
}

/** 用户 API Key 详情 */
export interface UserApiKeyDetail extends UserApiKey {
}

/** 创建用户 API Key 请求 */
export interface CreateUserApiKeyRequest {
  userId: number;
  applicationId: number;
  name: string;
}

/** 更新用户 API Key 请求 */
export interface UpdateUserApiKeyRequest {
  applicationId?: number;
  name?: string;
}

/** 创建用户 API Key 响应 */
export interface CreateUserApiKeyResponse {
  id: number;
  keyPrefix: string;
  keyPlain: string;
}
